packages/server/src/queries/complex/assets/price/index.ts

```diff
@@ -12,20 +12,39 @@ import { getPriceFromSidecar } from "./providers/sidecar";
 
 /** Provides a price (no caching) given a valid asset from asset list and a fiat currency code.
  *  @throws if there's an issue getting the price. */
-export type PriceProvider = (
-  asset: Asset,
-  currency?: CoingeckoVsCurrencies
-) => Promise<Dec>;
+export type PriceProvider = (coinMinimalDenom: string) => Promise<Dec>;
 
 const pricesCache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+
+/** Finds the fiat value of a single unit of a given asset denominated in USDC.
+ *  Assets can be identified either by `coinMinimalDenom`.
+ *  @throws if the coinMinimalDenom is not given or fails to fetch the price from data source */
+export async function getNativeAssetPrice(
+  coinMinimalDenom: string
+): Promise<Dec> {
+  if (!coinMinimalDenom) {
+    throw new Error("coinDenom is required");
+  }
+
+  return cachified({
+    key: `asset-price-nat-${coinMinimalDenom}`,
+    cache: pricesCache,
+    ttl: 1000 * 10, // 10 seconds
+    getFreshValue: () => getPriceFromSidecar(coinMinimalDenom),
+  });
+}
```

Comment:
@jonator instead of doing the factory pattern that we discussed, I simply introduced another helper.